tests/test_axicon.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+from scipy import special
+from scipy.constants import c, epsilon_0
+
+from lasy.laser import Laser
+from lasy.optical_elements import Axicon
+from lasy.profiles.gaussian_profile import GaussianProfile
+
+wavelength = 0.8e-6
+w0 = 10.0e-5  # m, initialized in near field
+pol = (1, 0)
+laser_energy = 1.0  # J
+t_peak = 0.0e-15  # s
+tau = 30.0e-15  # s
+gaussian_profile = GaussianProfile(wavelength, pol, laser_energy, w0, tau, t_peak)
+gamma = 2.2 * np.pi / 180
+z_max = w0 * np.cos(gamma) / np.sin(gamma)
+
+
+def check_axicon(laser):
+    distances = np.linspace(z_max / 1.5, z_max, 2)
+
+    errors = []
+    for dist in distances:
+        # create a new laser for each distance
+        new_laser = Laser(
+            laser.dim,
+            laser.grid.lo,
+            laser.grid.hi,
+            laser.grid.npoints,
+            gaussian_profile,
+        )
+
+        new_laser.apply_optics(Axicon(gamma=gamma))
+        new_laser.propagate(dist)
+        # Check the Bessel profile
+        error = check_bessel_profile(new_laser, dist)
+        errors.append(error)
+        print(errors)
+    assert np.max(errors) < 1e-2
+
+
+def check_bessel_profile(laser, z):
+    # Calculate the laser profile
+    field = laser.grid.get_temporal_field()
+    if laser.dim == "xyt":
+        Nx = field.shape[0]
+        A2 = 0.5 * epsilon_0 * c * (np.abs(field[Nx // 2 - 1, :, :]) ** 2).max(-1)
+        ax = laser.grid.axes[1]
+    else:
+        A2 = 0.5 * epsilon_0 * c * (np.abs(field[0, :, :]) ** 2).sum(-1)
+        ax = laser.grid.axes[0]
+        if ax[0] > 0:
+            A2 = np.r_[A2[::-1], A2]
+            ax = np.r_[-ax[::-1], ax]
+        else:
+            A2 = np.r_[A2[::-1][:-1], A2]
+            ax = np.r_[-ax[::-1][:-1], ax]
+
+    # Calculate the expected Bessel profile
+    k = 2 * np.pi / wavelength
+    r = np.abs(ax)
+    kr = k * np.sin(gamma) * r
+    power = laser_energy / (np.sqrt(np.pi / 2) * tau)
+
+    # Equation 3 in Brzobohaty, O., Cizmar, T., & Zemanek, P. (n.d.). High quality quasi-Bessel beam generated by round-tip axicon.
+    expected_profile = (
+        4
+        * power
+        * k
+        * np.sin(gamma)
+        / w0
+        * np.exp(-2 * z**2 / (z_max**2))
+        * z
+        / z_max
+        * special.j0(kr) ** 2
+    )
+
+    # Normalize profiles
+    A2_norm = A2 / np.max(A2)
+    expected_profile_norm = expected_profile / np.max(expected_profile)
+    # Calculate error
+    error1 = np.mean(np.abs(expected_profile_norm - A2_norm))
```

Comment:
It would make sense to use:
```suggestion
    error1 = np.max(np.abs(expected_profile - A2))/abs(A2).max()
```
But I think that the test fails in that case. Would you be able to check why?